main.go

```diff
@@ -113,29 +119,50 @@ func getEnvVarOrExit(name string) string {
 	return value
 }
 
-func main() {
-	// Initialize logging.
-	if err := logging.Initialize(os.Stdout, logging.FmtLogfmt, logging.LevelDebug, nil); err != nil {
-		fmt.Fprintf(os.Stderr, "ERROR: Unable to initialize logging: %v\n", err)
-		os.Exit(1)
-	}
-
-	// Get server port from environment variable or use the default.
-	port := os.Getenv(GatewayPortEnvVar)
-	if port == "" {
-		port = "8080"
+// Return the server port that should be used or exit if it is malformed.
+func getPortOrExit() int {
+	portStr := os.Getenv(GatewayPortEnvVar)
+	if portStr == "" {
+		portStr = "8080"
 	}
-	nPort, err := strconv.Atoi(port)
+	port, err := strconv.Atoi(portStr)
 	if err != nil {
 		logger.Error("malformed environment variable",
 			"err", err,
 			"name", GatewayPortEnvVar,
 		)
 		os.Exit(1)
 	}
+	return port
+}
+
+// Print version information.
+func printVersionInfo() {
+	fmt.Printf("Software version: %s\n", common.SoftwareVersion)
+	fmt.Printf("Rosetta API version: %s\n", common.RosettaAPIVersion)
+	fmt.Printf("Go toolchain version: %s\n", common.ToolchainVersion)
```

Comment:
Minor consistency nit, in Oasis Core we align (pad with spaces) all the versions in the same group.

Reply:
If there's an elegant way to get the Oasis Core version from `go.mod` (or similar), we could have something like:

```
Software version: 1.0.0
Oasis Core:
  Software version: 20.12.2
Rosetta API version: 1.4.1
Go toolchain version: 1.15.5
```
and later:
```
Software version: 1.0.0
Oasis Core:
  Software version:        20.12.2
  Client protocol version: 1.0.0
Rosetta API version: 1.4.1
Go toolchain version: 1.15.5
```

Or did you mean we should expose Oasis Core's consensus protocol versions here?

As for aligning the Software, Rosetta API and Go toolchain versions, I don't think we align them in Oasis Node either?

Reply:
> Or did you mean we should expose Oasis Core's consensus protocol versions here?

No, just the software version.

> As for aligning the Software, Rosetta API and Go toolchain versions, I don't think we align them in Oasis Node either?

Well we align in the same group as above (software and go toolchain versions are each in their own group in oasis-node).

Reply:
> > Or did you mean we should expose Oasis Core's consensus protocol versions here?
> 
> No, just the software version.

Ok, cool.

> > As for aligning the Software, Rosetta API and Go toolchain versions, I don't think we align them in Oasis Node either?
> 
> Well we align in the same group as above (software and go toolchain versions are each in their own group in oasis-node).

Aha, ok. I would prefer treating software, Rosetta API and Go toolchain as separate groups, similar to Oasis Node and other Oasis Core CLI tools.

Reply:
Sounds good.